internal/tools/orchestrator/services/addon/addon_handler.go

```diff
@@ -268,6 +270,21 @@ func (a *Addon) AddonDelete(req apistructs.AddonDirectDeleteRequest) error {
 	return nil
 }
 
+func generateTenantID(projectID string, tenantType, workspace string) string {
```

Comment:
Consider moving all callers to the `utils` package to eliminate duplicate logic.